client/components/data/query-post-stats/index.jsx

```diff
@@ -0,0 +1,67 @@
+/**
+ * External dependencies
+ */
+import { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import { isRequestingPostStat } from 'state/stats/posts/selectors';
+import { requestPostStat } from 'state/stats/posts/actions';
+
+class QueryPostStats extends Component {
+	componentWillMount() {
+		if ( !this.props.requestingPostStat &&
```

Comment:
What do you think about object destructuring from `this.props` here to eliminate the repeating of `this.props` so much?

`const { requestingPostStat, siteId, postId, stat } = this.props;`